x-pack/plugins/ml/public/jobs/new_job_new/common/job_creator/population_job_creator.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { JobCreator } from './job_creator';
+import { Field, Aggregation, SplitField } from '../../../../../common/types/fields';
+import { Detector } from './configs';
+import { createBasicDetector } from './util/default_configs';
+
+export class PopulationJobCreator extends JobCreator {
+  // a population job has one overall over (split) field, which is the same for all detectors
+  // each detector has an optional by field
+  private _splitField: SplitField = null;
+  private _byFields: SplitField[] = [];
+
+  // add a by field to a specific detector
+  public setByField(field: SplitField, index: number) {
+    if (field === null) {
+      this.removeByField(index);
+    } else {
+      if (this._detectors[index] !== undefined) {
+        this._byFields[index] = field;
+        this._detectors[index].by_field_name = field.id;
+      }
+    }
+  }
+
+  // remove a by field from a specific detector
+  public removeByField(index: number) {
+    if (this._detectors[index] !== undefined) {
+      this._byFields[index] = null;
+      delete this._detectors[index].by_field_name;
+    }
+  }
+
+  // get the by field for a specific detector
+  public getByField(index: number): SplitField {
+    if (this._byFields[index] === undefined) {
+      return null;
+    }
+    return this._byFields[index];
```

Comment:
Couldn't this be `return (this._byFields[index] || null)`? If you're depending on `null` being returned.

Reply:
Yes it could, but I quite like the explicitness of this check. that it's checking for `undefined` and then returning `null` if it finds it.

Reply:
Yeah definitely a better solution if we only want `undefined` to be replaced - cool 👌